### PR TITLE
Update artifacts-common entra user service connection 

### PR DESCRIPTION
**Pull request by embetten**
- Add retries and ppe login

### CODE REVIEW

File: common-npm-packages/artifacts-common/EntraWifUserServiceConnectionUtils.ts [hunk 1]
```diff
@@ -2,76 +2,96 @@ import path = require("path");
...
-    try {
-        let tenant = tenantId ?? tl.getEndpointAuthorizationParameterRequired(serviceConnectionName, "TenantId");
-        tl.debug(tl.loc('Info_UsingTenantId', tenantId));
-        const systemAccessToken = getSystemAccessToken();
-        const url = process.env["SYSTEM_OIDCREQUESTURI"]+"?api-version=7.1&serviceConnectionId="+serviceConnectionName;
-        
-        const ADOResponse: {oidcToken: string} = await (await fetch(url, 
-        {
+export async function getFederatedWorkloadIdentityCredentials(serviceConnectionName: string, tenantId?: string) : Promise<string | undefined>{
+    let tenant = tenantId ?? tl.getEndpointAuthorizationParameterRequired(serviceConnectionName, "TenantId");
```
Comment (AlexVTor) [inline common-npm-packages/artifacts-common/EntraWifUserServiceConnectionUtils.ts:15]: getEndpointAuthorizationParameterRequired() will throw if the required parameter isn't set and has now been moved outside of the try block. Maybe we should add a comment so callers are aware of this and will handle the throw?

File: common-npm-packages/artifacts-common/EntraWifUserServiceConnectionUtils.ts [hunk 2]
```diff
@@ -2,76 +2,96 @@ import path = require("path");
...
                 'Content-Type': 'application/json',
                 'Authorization': 'Bearer '+ systemAccessToken
             }
-        })).json() as {oidcToken: string};
+        }).then(async response => {
+            var oidcObject = await (response?.json()) as {oidcToken: string};
 
-        tl.setSecret(ADOResponse.oidcToken);
-        let entraURI = "https://login.windows.net/"+tenant+"/oauth2/v2.0/token";
+            if (!oidcObject?.oidcToken){
```
Comment (AlexVTor) [inline common-npm-packages/artifacts-common/EntraWifUserServiceConnectionUtils.ts:30]: Out of curiosity, what is oidc? It looks like it's part of the flow to gather tokens but curious to hear more about it

Comment (embetten) [reply common-npm-packages/artifacts-common/EntraWifUserServiceConnectionUtils.ts:30]: OIDC (short for openID Connect) is a protocol built on top of the Oauth protocol. It allows the federated credential flow. More info here: https://learn.microsoft.com/en-us/entra/identity-platform/v2-protocols. I can provide more resources offline too.

File: common-npm-packages/artifacts-common/EntraWifUserServiceConnectionUtils.ts (hunk 1 above)
Comment (embetten) [reply common-npm-packages/artifacts-common/EntraWifUserServiceConnectionUtils.ts:15]: It prints a nice error message with it already! I am planning to leave as is.

**Review by AlexVTor — APPROVED** [review]

File: common-npm-packages/artifacts-common/EntraWifUserServiceConnectionUtils.ts [hunk 3]
```diff
@@ -2,76 +2,96 @@ import path = require("path");
...
+        });
+    }, 3, 1000);
 }
 
-export async function getFeedTenantId(feedUrl: string) : Promise<string | undefined>
-{
-    try
-    {
+export async function getFeedTenantId(feedUrl: string) : Promise<string | undefined>{
+    try {
```
[the remark this answers is not in the document]
Comment (embetten) [reply common-npm-packages/artifacts-common/EntraWifUserServiceConnectionUtils.ts:73]: No, and feed url should be best effort anyway - it falls back to the tenant in the service connection. We can add later if it becomes an issue.

**Pull request merged**